script.js

```diff
@@ -29,7 +39,55 @@ async function fetchQuizzes() {
     }, { quizzes: [], threeStrikes: [], hardWords: [] });
 }
 
+function renderQuizRow(id, title, link, complete = false, score = 0) {
+  const date = /:\s(?<date>[a-z]+\s[0-9]+,\s[0-9]+)/i.exec(title)?.groups?.date;
+  const name = /:\s(?<name>(?:Morning|Afternoon)\s[a-z\s]+):/i.exec(title)?.groups?.name;
+  const row = document.createElement("tr");
+  const td1 = document.createElement("td");
+  const label = document.createElement("label");
+  label.setAttribute("for", id);
+  const checkbox = document.createElement("input");
+  checkbox.setAttribute("type", "checkbox");
+  checkbox.checked = complete;
+  checkbox.id = id;
+  checkbox.name = id;
+  checkbox.oninput = () => { void markComplete(id, checkbox.checked); };
+  label.appendChild(checkbox);
+  td1.appendChild(label);
+  const td2 = document.createElement("td");
+  const anchor = document.createElement("a");
+  anchor.href = link;
+  anchor.target = "_blank";
+  anchor.innerText = name;
+  td2.appendChild(anchor);
+  const td3 = document.createElement("td");
+  td3.innerText = date;
+  const td4 = document.createElement("td");
+  const input = document.createElement("input");
+  input.setAttribute("type", "number");
+  input.setAttribute("step", "1");
+  input.value = score;
+  input.oninput = () => { void setScore(id, parseInt(input.value)); };
```

Comment:
damn. First time I see `void` being used this way. What does it do here?

Reply:
In this case, it explicitly ignores the return value from the async function - avoids scope colouring (or whatever it's called). So instead of:
```ts
input.oninput = async () => { await setScore(id, parseInt(input.value)); };
```
You can void the call, because the `await` is useless.

Reply:
Aha! It's called "function colouring". I was close lol